File: app/src/main/java/com/example/intent_aplication/MainActivity.java
package com.example.intent_aplication;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.Toast;


import java.text.DecimalFormat;

public class MainActivity extends AppCompatActivity {
    TextView textoFinal, txtmedida, textMedida1, textMedida2;
    EditText valor, indice;
    Button btnCalcular;
    ImageView image, imageWhats, imageMessenger, imageGmail;
    Spinner spnlista;
    Intent intent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        spnlista = new Spinner(this);
        image = new ImageView(this);
        textoFinal = new TextView(this);

        valor = findViewById(R.id.editText1);

        imageWhats = findViewById(R.id.imageWhats);
        imageGmail = findViewById(R.id.imageGmail);
        imageMessenger = findViewById(R.id.imageMessernger);
        btnCalcular = findViewById(R.id.btnCalcular);
        spnlista = findViewById(R.id.spinner);
        image = findViewById(R.id.imageView2);
        txtmedida = findViewById(R.id.txtInforma);
        textMedida1 = findViewById(R.id.textView);
        textMedida2 = findViewById(R.id.textView2);


        spnlista.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {

            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {

                if (position == 0){
                    image.setImageResource(R.drawable.polegada);
                    textMedida1.setText(R.string.polegada);
                    textMedida2.setText(R.string.centimetro);
                }
                else if (position == 1) {
                    image.setImageResource(R.drawable.medida_altura);
                    textMedida1.setText(R.string.pes);
                    textMedida2.setText(R.string.metros);

                }
                else if (position == 2){
                    image.setImageResource(R.drawable.quilometragem);
                    txtmedida.setText(R.string.milha);
                    textMedida1.setText(R.string.milha);
                    textMedida2.setText(R.string.quilometro);
                }
                else if (position == 3){
                    image.setImageResource(R.drawable.temperatura_calor_capa);
                    textMedida1.setText(R.string.celsius);
                    textMedida2.setText(R.string.farenheint);
                }
                else if (position == 4){
                    image.setImageResource(R.drawable.nos);
                    textMedida1.setText(R.string.nos);
                    textMedida2.setText(R.string.km_h);
                }
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {

            }
        });
    }

    public void btnLimpar(View view){
        valor = findViewById(R.id.editText1);
        txtmedida = findViewById(R.id.txtInforma);

        valor.setText("");
        txtmedida.setText(R.string.valor_vazio);
    }

    public void btnTrocar(View view){
        valor = findViewById(R.id.editText1);
        textMedida1 = findViewById(R.id.textView);
        textMedida2 = findViewById(R.id.textView2);

        String med_1 = textMedida1.getText().toString();
        String med_2 = textMedida2.getText().toString();

        textMedida1.setText(med_2);
        textMedida2.setText(med_1);

        String entrada = valor.getText().toString();
        conversao(entrada);
    }

    public void btnCalcular(View view) {
        valor = findViewById(R.id.editText1);
        txtmedida = findViewById(R.id.txtInforma);
        String entrada = valor.getText().toString();

        conversao(entrada);
    }

    public void conversao(String entrada){
            txtmedida = findViewById(R.id.txtInforma);
            if (entrada.equals(""))
                txtmedida.setText(R.string.valor_vazio);
            else {
                float numero = Float.parseFloat(valor.getText().toString());
                if (textMedida1.getText().toString().equals("Polegada")){
                    double resultado = numero * 2.54;
                    DecimalFormat formatado = new DecimalFormat("0.00");
                    String textResultado = formatado.format(resultado) + " Centímetros" + "Unidade de Medida - SI";
                    txtmedida.setText(textResultado);
                }
                else if (textMedida1.getText().toString().equals("Centímetro")){
                    double resultado = numero * 0.39;
                    DecimalFormat formatado = new DecimalFormat("0.00");
                    String textResultado = formatado.format(resultado) + " Polegadas" + "Unidade de Medida - EUA";
                    txtmedida.setText(textResultado);
                }
                else if (textMedida1.getText().toString().equals("Pés")){
                    double resultado = numero * 0.3;
                    DecimalFormat formatado = new DecimalFormat("0.00");
                    String textResultado = formatado.format(resultado) + " Metros" + "Unidade de Medida - SI";
                    txtmedida.setText(textResultado);
                }
                else if (textMedida1.getText().toString().equals("Metros")){
                    double resultado = numero * 3.28;
                    DecimalFormat formatado = new DecimalFormat("0.00");
                    String textResultado = formatado.format(resultado) + " Pés" + "SI Antiga - EUA";
                    txtmedida.setText(textResultado);
                }
                else if (textMedida1.getText().toString().equals("Milha")){
                    double resultado = numero * 1.6;
                    DecimalFormat formatado = new DecimalFormat("0.00");
                    String textResultado = formatado.format(resultado) + " Quilometros" + " Unidade de Medida SI";
                    txtmedida.setText(textResultado);
                }
                else if (textMedida1.getText().toString().equals("Quilometros")){
                    double resultado = numero * 0.62;
                    DecimalFormat formatado = new DecimalFormat("0.00");
                    String textResultado = formatado.format(resultado) + " Milhas" + "SI Antiga - EUA";
                    txtmedida.setText(textResultado);
                }
                else if (textMedida1.getText().toString().equals("Celsius")){
                    double resultado = (numero * 1.8) + 32;
                    DecimalFormat formatado = new DecimalFormat("0.00");
                    String textResultado = formatado.format(resultado) + "ºF" + "SI Antiga - EUA";
                    txtmedida.setText(textResultado);
                }
                else if (textMedida1.getText().toString().equals("Farenheint")){
                    double resultado = ((numero -32) * 5) / 9;
                    DecimalFormat formatado = new DecimalFormat("0.00");
                    String textResultado = formatado.format(resultado) + "ºC" + " Unidade de Medida - SI";
                    txtmedida.setText(textResultado);
                }
                else if (textMedida1.getText().toString().equals("Nós")){
                    double resultado = numero * 1.8;
                    DecimalFormat formatado = new DecimalFormat("0.00");
                    String textResultado = formatado.format(resultado) + " Km/h" + " Unidade de Medida - SI";
                    txtmedida.setText(textResultado);
                }
                else if (textMedida1.getText().toString().equals("Km/h")){
                    double resultado = numero * 0.5;
                    DecimalFormat formatado = new DecimalFormat("0.00");
                    String textResultado = formatado.format(resultado) + " Nós" + " SI Antiga - EUA";
                    txtmedida.setText(textResultado);
                }
        }
    }

    public void btnEnviaWhatsapp(View view){
        valor = findViewById(R.id.editText1);
        String num = valor.getText().toString();
        if (! num.equals("")){
            PackageManager pm = getPackageManager();
            try {
                Intent intent = new Intent(Intent.ACTION_SEND);
                intent.setType("text/plain");
                intent.putExtra(Intent.EXTRA_TEXT, txtmedida.getText().toString());
                PackageInfo info = pm.getPackageInfo("com.whatsapp", PackageManager.GET_META_DATA);
                intent.setPackage("com.whatsapp");
                startActivity(intent);
            } catch (PackageManager.NameNotFoundException e) {
                Toast.makeText(this, "Não foi possível enviar a mensagem", Toast.LENGTH_LONG).show();
            }
        }
       else{
           Toast.makeText(this, "Digite algum valor na caixa de texto", Toast.LENGTH_LONG).show();
        }
    }

    public void btnEnviaGoogleDocs(View view) {
        valor = findViewById(R.id.editText1);
        String num = valor.getText().toString();
        if (! num.equals("")){
            PackageManager pm = getPackageManager();
            try {
                Intent intent = new Intent(Intent.ACTION_SEND);
                intent.setType("text/plain");
                intent.putExtra(Intent.EXTRA_TEXT, txtmedida.getText().toString());
                intent.putExtra(Intent.EXTRA_SUBJECT, "Nota da atividade Intents - ILP 506");
                PackageInfo info = pm.getPackageInfo("com.google.android.apps.docs", PackageManager.GET_META_DATA);
                intent.setPackage("com.google.android.apps.docs");
                startActivity(intent);
            } catch (PackageManager.NameNotFoundException e) {
                Toast.makeText(this, "Não foi possível enviar a mensagem", Toast.LENGTH_LONG).show();
            }
        }
        else{
            Toast.makeText(this, "Digite algum valor na caixa de texto", Toast.LENGTH_LONG).show();
        }

    }

    public void btnEnviaGmail(View view) {
        valor = findViewById(R.id.editText1);
        String num = valor.getText().toString();
        if (! num.equals("")){
            PackageManager pm = getPackageManager();
            try {
                Intent intent = new Intent(Intent.ACTION_SEND);
                intent.setType("message/rfc822");
                intent.putExtra(Intent.EXTRA_EMAIL, "christian.cubo@fatec.gov.sp.br");
                intent.putExtra(Intent.EXTRA_SUBJECT, "Nota da atividade Intents - Matéria ILP 506");
                intent.putExtra(Intent.EXTRA_TEXT, txtmedida.getText().toString());
                PackageInfo info = pm.getPackageInfo("com.google.android.gm", PackageManager.GET_META_DATA);
                intent.setPackage("com.google.android.gm");
                startActivity(Intent.createChooser(intent, "Send Email"));
            } catch (PackageManager.NameNotFoundException e) {
                Toast.makeText(this, "Não foi possível enviar a mensagem", Toast.LENGTH_LONG).show();
            }
        }
        else{
            Toast.makeText(this, "Digite algum valor na caixa de texto", Toast.LENGTH_LONG).show();
        }
    }
}

File: app/src/main/java/com/example/conversordemedidas/MainActivity.kt
package com.example.conversordemedidas

import android.content.Intent
import android.content.pm.PackageManager
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.EditText
import android.widget.ImageView
import android.widget.Spinner
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import java.text.DecimalFormat

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Spinner(this)
        ImageView(this)
        TextView(this)
        val spnlista = findViewById<Spinner>(R.id.spinner)
        val image = findViewById<ImageView>(R.id.imageView2)
        val txtmedida = findViewById<TextView>(R.id.txtInforma)
        val textMedida1 = findViewById<TextView>(R.id.textView)
        val textMedida2 = findViewById<TextView>(R.id.textView2)
        spnlista.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(
                parent: AdapterView<*>?,
                view: View,
                position: Int,
                id: Long
            ) {
                when (position) {
                    0 -> {
                        image.setImageResource(R.drawable.imagem_sistemas_medidas)
                        textMedida1.text = ""
                        textMedida2.text = ""
                    }
                    1 -> {
                        image.setImageResource(R.drawable.polegada)
                        textMedida1.setText(R.string.polegada)
                        textMedida2.setText(R.string.centimetro)
                    }
                    2 -> {
                        image.setImageResource(R.drawable.medida_altura)
                        textMedida1.setText(R.string.pes)
                        textMedida2.setText(R.string.metros)
                    }
                    3 -> {
                        image.setImageResource(R.drawable.quilometragem)
                        txtmedida.setText(R.string.milha)
                        textMedida1.setText(R.string.milha)
                        textMedida2.setText(R.string.quilometro)
                    }
                    4 -> {
                        image.setImageResource(R.drawable.temperatura_calor_capa)
                        textMedida1.setText(R.string.celsius)
                        textMedida2.setText(R.string.farenheint)
                    }
                    5 -> {
                        image.setImageResource(R.drawable.nos)
                        textMedida1.setText(R.string.nos)
                        textMedida2.setText(R.string.km_h)
                    }
                }
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {}
        }
    }

    fun btnLimpar(view: View?) {
        val valor = findViewById<EditText>(R.id.editText1)
        val txtmedida = findViewById<TextView>(R.id.txtInforma)
        valor.setText("")
        txtmedida.setText(R.string.valor_vazio)
    }

    fun btnTrocar(view: View?) {
        val valor = findViewById<EditText>(R.id.editText1)
        val textMedida1 = findViewById<TextView>(R.id.textView)
        val textMedida2 = findViewById<TextView>(R.id.textView2)
        val med1 = textMedida1.text.toString()
        val med2 = textMedida2.text.toString()
        textMedida1.text = med2
        textMedida2.text = med1
        val entrada = valor.text.toString()
        conversao(entrada)
    }

    fun btnCalcular(view: View?) {
        val valor = findViewById<EditText>(R.id.editText1)
        val entrada = valor.text.toString()
        conversao(entrada)
    }

    fun conversao(entrada: String) {
        val txtmedida = findViewById<TextView>(R.id.txtInforma)
        val textMedida1 = findViewById<TextView>(R.id.textView)
        if (entrada == "") txtmedida.setText(R.string.valor_vazio) else {
            var numero = entrada.toFloat()
            if (textMedida1.text.toString() == "Polegada") {
                val resultado = numero * 2.54
                val formatado = DecimalFormat("0.00")
                val textResultado =
                    formatado.format(resultado) + " Centímetros" + "\n Unidade de Medida - SI"
                txtmedida.text = textResultado
            } else if (textMedida1.text.toString() == "Centímetro") {
                val resultado = numero * 0.39
                val formatado = DecimalFormat("0.00")
                val textResultado =
                    formatado.format(resultado) + " Polegadas" + "\n Unidade de Medida - EUA"
                txtmedida.text = textResultado
            } else if (textMedida1.text.toString() == "Pés") {
                val resultado = numero * 0.3
                val formatado = DecimalFormat("0.00")
                val textResultado =
                    formatado.format(resultado) + " Metros" + "\n Unidade de Medida - SI"
                txtmedida.text = textResultado
            } else if (textMedida1.text.toString() == "Metros") {
                val resultado = numero * 3.28
                val formatado = DecimalFormat("0.00")
                val textResultado = formatado.format(resultado) + " Pés" + "\n SI Antiga - EUA"
                txtmedida.text = textResultado
            } else if (textMedida1.text.toString() == "Milha") {
                val resultado = numero * 1.6
                val formatado = DecimalFormat("0.00")
                val textResultado =
                    formatado.format(resultado) + " Quilometros" + "\n Unidade de Medida SI"
                txtmedida.text = textResultado
            } else if (textMedida1.text.toString() == "Quilometros") {
                val resultado = numero * 0.62
                val formatado = DecimalFormat("0.00")
                val textResultado = formatado.format(resultado) + " Milhas" + "\n SI Antiga - EUA"
                txtmedida.text = textResultado
            } else if (textMedida1.text.toString() == "Celsius") {
                val resultado = numero * 1.8 + 32
                val formatado = DecimalFormat("0.00")
                val textResultado = formatado.format(resultado) + "ºF" + "\n SI Antiga - EUA"
                txtmedida.text = textResultado
            } else if (textMedida1.text.toString() == "Farenheint") {
                val resultado = ((numero - 32) * 5 / 9).toDouble()
                val formatado = DecimalFormat("0.00")
                val textResultado = formatado.format(resultado) + "ºC" + "\n Unidade de Medida - SI"
                txtmedida.text = textResultado
            } else if (textMedida1.text.toString() == "Nós") {
                val resultado = numero * 1.8
                val formatado = DecimalFormat("0.00")
                val textResultado =
                    formatado.format(resultado) + " Km/h" + "\n Unidade de Medida - SI"
                txtmedida.text = textResultado
            } else if (textMedida1.text.toString() == "Km/h") {
                val resultado = numero * 0.5
                val formatado = DecimalFormat("0.00")
                val textResultado = formatado.format(resultado) + " Nós" + "\n SI Antiga - EUA"
                txtmedida.text = textResultado
            }
        }
    }

    fun btnEnviaWhatsapp(view: View?) {
        val valor = findViewById<EditText>(R.id.editText1)
        val txtmedida = findViewById<TextView>(R.id.txtInforma)
        val num = valor.text.toString()
        if (num != "") {
            val pm = packageManager
            try {
                val intent = Intent(Intent.ACTION_SEND)
                intent.setType("text/plain")
                intent.putExtra(Intent.EXTRA_TEXT, txtmedida!!.text.toString())
                intent.setPackage("com.whatsapp")
                startActivity(intent)
            } catch (e: PackageManager.NameNotFoundException) {
                Toast.makeText(this, "Não foi possível enviar a mensagem", Toast.LENGTH_SHORT).show()
            }
        } else {
            Toast.makeText(this, "Digite algum valor na caixa de texto", Toast.LENGTH_SHORT).show()
        }
    }

    fun btnEnviaGoogleDocs(view: View?) {
        val valor = findViewById<EditText>(R.id.editText1)
        val txtmedida = findViewById<TextView>(R.id.txtInforma)
        val num = valor.text.toString()
        if (num != "") {
            val pm = packageManager
            try {
                val intent = Intent(Intent.ACTION_SEND)
                intent.setType("text/plain")
                intent.putExtra(Intent.EXTRA_TEXT, txtmedida!!.text.toString())
                intent.putExtra(Intent.EXTRA_SUBJECT, "Nota da atividade Intents - ILP 506")
                intent.setPackage("com.google.android.apps.docs")
                startActivity(intent)
            } catch (e: PackageManager.NameNotFoundException) {
                Toast.makeText(this, "Não foi possível enviar a mensagem", Toast.LENGTH_SHORT).show()
            }
        } else {
            Toast.makeText(this, "Digite algum valor na caixa de texto", Toast.LENGTH_SHORT).show()
        }
    }

    fun btnEnviaGmail(view: View?) {
        val valor = findViewById<EditText>(R.id.editText1)
        val txtmedida = findViewById<TextView>(R.id.txtInforma)
        val num = valor.text.toString()
        if (num != "") {
            val pm = packageManager
            try {
                val intent = Intent(Intent.ACTION_SEND)
                intent.type = "text/plain"
                intent.putExtra(Intent.EXTRA_EMAIL, arrayOf("chriscubo@gmail.com"))
                intent.putExtra(Intent.EXTRA_SUBJECT, "Nota da atividade Intents - Matéria ILP 506")
                intent.putExtra(Intent.EXTRA_TEXT, txtmedida?.text.toString())
                intent.setPackage("com.google.android.gm")
                startActivity(Intent.createChooser(intent, "Send Email"))
            } catch (e: PackageManager.NameNotFoundException) {
                Toast.makeText(this, "Não foi possível enviar a mensagem", Toast.LENGTH_SHORT).show()
            }
        } else {
            Toast.makeText(this, "Digite algum valor na caixa de texto", Toast.LENGTH_SHORT).show()
        }
    }
}
